deepinv/training_utils.py

```diff
@@ -183,7 +203,12 @@ def train(
                     )  # In this case the dataloader outputs also a class label
                     x = x.to(device)
                     physics_cur = physics[g]
-                    physics_cur.reset()
+
+                    if isinstance(physics_cur, torch.nn.DataParallel):
+                        physics_cur.module.noise_model.__init__(sigma_max=0.2, x=x)
```

Comment:
I do not get why this ?

Reply:
In this case, `physics_cur` needs to be re-instanciated because it's not possible to update physics parameters within the forward pass in a `nn.DataParallel` module, they need to be reset instead.